File: mono_1/src/functions/learningAutomaton.jl
function automaton_F1(prob::Array, alpha::Float64, beta::Int64, pos::Int64)

    nova_prob = prob[pos] + (alpha * beta * (1 - prob[pos])) - (alpha * (1 - beta) * prob[pos])
    prob[pos] = nova_prob

end

function automaton_F2(prob::Array, alpha::Float64, beta::Int64, pos::Int64)

    nova_prob = prob[pos] - (alpha * beta * prob[pos]) + (alpha * (1 - beta) * ((1 / (length(prob) - 1)) - prob[pos]))
    prob[pos] = nova_prob

end

File: mono_2/src/heuristics/learning_automaton.py
def automaton_F1(prob, alpha, beta, pos):

    nova_prob = prob[pos] + (alpha * beta * (1 - prob[pos])) - (alpha * (1 - beta) * prob[pos])
    prob[pos] = nova_prob

def automaton_F2(prob, alpha, beta, pos):

    nova_prob = prob[pos] - (alpha * beta * prob[pos]) + (alpha * (1 - beta) * ((1 / (len(prob) - 1)) - prob[pos]))
    prob[pos] = nova_prob